DNN Platform/Library/Startup.cs

```diff
@@ -26,6 +30,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<IDnnContext, DotNetNukeContext>();
 
             services.AddTransient(x => PortalController.Instance);
+            services.AddTransient<INewHostController, HostController>();

```

Comment:
This should be registered as a scoped service and not Transient. Scoped services are re-used for the life of the request, where transient is a new instance every time requested.

Reply:
This has been resolved in the latest code push and it now uses `Scoped`